### ISSUE TITLE
Python 3.10: TypeError: PurePath._from_parts() got an unexpected keyword argument 'init'

### ISSUE THREAD

**Issue by hroncok** [facelessuser/wcmatch#176]
Hello, in Fedora we see test failures with Python 3.10.0b1:

```
...
______________________ TestCWD.test_cwd_root_dir_pathlike ______________________
self = <tests.test_glob.TestCWD object at 0x7f4b20972bf0>
    def test_cwd_root_dir_pathlike(self):
        """Test root level glob when we switch directory via `root_dir` with a path-like object."""
    
>       self.assert_equal(glob.glob('EF', root_dir=pathlib.Path(self.tempdir)), ['EF'])
tests/test_glob.py:1136: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
cls = <class 'wcmatch.pathlib.PosixPath'>, args = ('@test_188_tmp_dir',)
kwargs = {}
    def __new__(cls, *args, **kwargs):
        """New."""
    
        if cls is Path:
            cls = WindowsPath if os.name == 'nt' else PosixPath
>       self = cls._from_parts(args, init=False)
E       TypeError: PurePath._from_parts() got an unexpected keyword argument 'init'
wcmatch/pathlib.py:80: TypeError
...
FAILED tests/test_glob.py::TestCWD::test_cwd_root_dir_pathlike - TypeError: P...
FAILED tests/test_globmatch.py::TestGlobMatchSpecial::test_filter_root_dir_pathlib
FAILED tests/test_globmatch.py::TestGlobMatchSpecial::test_match_root_dir_pathlib
FAILED tests/test_pathlib.py::TestGlob::test_glob - TypeError: PurePath._from...
FAILED tests/test_pathlib.py::TestGlob::test_integrity - TypeError: PurePath....
FAILED tests/test_pathlib.py::TestGlob::test_relative - TypeError: PurePath._...
FAILED tests/test_pathlib.py::TestGlob::test_relative_exclude - TypeError: Pu...
FAILED tests/test_pathlib.py::TestGlob::test_rglob - TypeError: PurePath._fro...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case0] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case1] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case2] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case3] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case4] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibGlobmatch::test_cases[case5] - TypeE...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case0] - TypeError...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case1] - TypeError...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case2] - TypeError...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case3] - TypeError...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case4] - TypeError...
FAILED tests/test_pathlib.py::TestPathlibMatch::test_cases[case5] - TypeError...
FAILED tests/test_pathlib.py::TestExceptions::test_absolute_glob - TypeError:...
FAILED tests/test_pathlib.py::TestExceptions::test_bad_path - TypeError: Pure...
FAILED tests/test_pathlib.py::TestExceptions::test_inverse_absolute_glob - Ty...
FAILED tests/test_pathlib.py::TestComparisons::test_equal - TypeError: PurePa...
FAILED tests/test_pathlib.py::TestComparisons::test_flavour_equal - TypeError...
FAILED tests/test_pathlib.py::TestComparisons::test_instance - TypeError: Pur...
FAILED tests/test_pathlib.py::TestComparisons::test_pickle - TypeError: PureP...
FAILED tests/test_pathlib.py::TestExpansionLimit::test_limit_glob - TypeError...
FAILED tests/test_pathlib.py::TestExpansionLimit::test_limit_rglob - TypeErro...
========== 29 failed, 1112 passed, 148 skipped, 1 deselected in 4.35s ==========

```

See https://bugs.python.org/issue40038 and https://github.com/python/cpython/commit/2219187cab6bca009c42b63b2f4c30b5b10c916d

**Comment by facelessuser** [comment]
Yeah, hadn't got around to testing 3.10 yet. I was vaguely aware that there was a little refactoring in pathlib, so this doesn't surprise me.

**Comment by facelessuser** [comment]
I have a fix in #177. I like that Py3.10 cleans this area up.

**Comment by facelessuser** [comment]
Tests on `master` pass now. While I won't formally say that I support Python 3.10 (as things could still change) the next release should not have problems with the beta 1 release. Do you guys need a release sooner rather than later?
